src/main/java/kenigsberg/weather/WeatherFrame.java

```diff
@@ -8,11 +8,79 @@
 
 import javax.swing.*;
 import java.awt.*;
-import java.io.IOException;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
 
 public class WeatherFrame extends JFrame {
-    CurrentWeather currentWeather;
+    CurrentWeatherView currentWeatherView;
+
+    public WeatherFrame() {
+
+        setSize(800, 600);
+        setTitle("Current Weather");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        JPanel mainPanel = new JPanel();
+        mainPanel.setLayout(new BorderLayout());
+
+        JPanel topPanel = new JPanel();
+        topPanel.setLayout(new FlowLayout());
+
+        JTextField location = new JTextField("Staten Island");
+        topPanel.add(location);
+
+        JButton button = new JButton("Submit");
+        topPanel.add(button);
+
+        mainPanel.add(topPanel, BorderLayout.NORTH);
+
+        currentWeatherView = new CurrentWeatherView();
+        mainPanel.add(currentWeatherView, BorderLayout.CENTER);
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .build();
+
 
+        WeatherService service = retrofit.create(WeatherService.class);
+
+
+        Disposable disposable = service.getFiveDayWeather("Staten Island")
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.newThread())
+                .subscribe(
+                        fiveDayWeather -> {
+                            currentWeatherView.setForecast(fiveDayWeather);
+                        },
+                        Throwable::printStackTrace
+
+                );
```

Comment:
Can you remove the code duplication between here and line 65?

Reply:
How can I do that? Is there a way to call the already-created subscribe?

Reply:
No. You won't call the already created subscribe.

You want the same code to be run in two places.
How can you do that?

Reply:
Oh! To create a method with that code? And call it where needed?

Reply:
yes